src/utils/httpTransport.ts

```diff
@@ -0,0 +1,65 @@
+enum METHOD {
+  GET = 'GET',
+  POST = 'POST',
+  PUT = 'PUT',
+  DELETE = 'DELETE'
+}
+
+type Options = {
+  method: METHOD;
+  data?: any;
+};
+
+type OptionsWithoutMethod = Omit<Options, 'method'>;
+
+function queryStringify(data: object) {
+  const keys = Object.keys(data);
+
+  return keys.reduce((result, key, index) => {
+    const value = data[key];
+    return `${result}${key}=${value}${index < keys.length - 1 ? '&' : ''}`;
+  }, '?');
+}
+
+export default class HTTPTransport {
+  get(url: string, options: OptionsWithoutMethod = {}): Promise<XMLHttpRequest> {
```

Comment:
Было бы круто сделать метод дженериком, чтобы можно было бы указывать тип возвращаемого значения извне
```suggestion
  get<T>(url: string, options: OptionsWithoutMethod = {}): Promise<T> {
```